frontend/src/layouts/AddressSettings.vue

```diff
@@ -0,0 +1,39 @@
+<template>
+  <div>
+    <connect-wallet>
+      <span v-if="userDisplayAddress" class="text-caption cursor-pointer dark-toggle">
+        {{ userDisplayAddress }}
+      </span>
+      <span v-else-if="!isSupportedNetwork" class="text-caption text-grey cursor-pointer dark-toggle">
+        <q-icon name="fas fa-exclamation-triangle" color="warning" left /> Unsupported network
+      </span>
+    </connect-wallet>
+    <span v-if="advancedMode" class="q-ml-md">
+      🧙 <q-tooltip content-class="bg-muted dark-toggle shadow-2 q-pa-md"> Advanced mode is on </q-tooltip>
+    </span>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from '@vue/composition-api';
+import ConnectWallet from 'components/ConnectWallet.vue';
+
+export default defineComponent({
+  name: 'AddressSettings',
+  components: { ConnectWallet },
+  props: {
+    userDisplayAddress: {
+      type: String,
+      required: false,
```

Comment:
I think this should be required, but with `null` or `undefined` as acceptable types

Reply:
But, what if the user has not connected their wallet yet?

Reply:
You'll see in `wallet.ts` where `userAddress` is initialized/defined, and it defaults to `undefined` if a user has not connected their wallet yet. So the prop would still be passed, just with a value of undefined